### ISSUE TITLE
Add example without using Iron Router

### ISSUE THREAD

**Issue by jaredtmartin** [pascoual/meteor-pdfkit#13]
The example given is great, but I'm trying to figure out how to use it without Iron Router. Could I get a simple example of that?

**Comment by pascoual** [comment]
You can use CollectionFS to store your created PDF and serve it to your client side:
https://github.com/CollectionFS/Meteor-CollectionFS

**Comment by jaredtmartin** [comment]
Actually, I've been wanting to do that too, but if I build the PDF on the server, I can't figure out how to import it into CollectionFS. If I could, that would be amazing!

**Comment by pascoual** [comment]
Yes you can. Take a look at this CollectionFS sample:
https://github.com/CollectionFS/Meteor-CollectionFS/issues/281

**Comment by jaredtmartin** [comment]
Wow, I have looked for that many times, but never saw it. 
They did something like this:
   if future.wait()
        fileObj = new FS.File(filePath)
        fileObj.owner = Meteor.userId()
        fileObj.metadata = orderId: orderId
        console.log FileStorage.insert fileObj

How can I have a future returned when the writesync function finishes?

Also: is there a way I can stream the file directly to CollectionFS?

I havn’t seen any of these common use cases demonstrated in the docs.  As a matter of fact, I don’t think writeSync appears anywhere in the docs.  Perhaps we could add these? If not, I could make a tutorial to help out others who are just trying to get started.

**Comment by pascoual** [comment]
Slava talk about this here:
http://stackoverflow.com/a/26601469

**Comment by conorstrejcek** [comment]
I've been looking into a similar problem which I have had using `writeSync`. I use `pdfkit` to generate a pdf file, and then when the file is completely written I need a callback to display the pdf. Right now there doesn't seem to be an easy way to get a callback from writeSync, so there is no way for me to check that the file has been completely written.

Without using something like CollectionFS or something similar, is there an easy way to wait for `writeSync` to finish writing?

**Comment by pascoual** [comment]
writeSync is sync, that mean that when next line is executed, file is already write (finished).
(if not, it's a bug)

**Comment by jaredtmartin** [comment]
Is there a writeAsync function? Neither writeAsync nor writeSync appear in the pdfkit docs. I imagine this is because you added the writeSync function.
I'm sorry if I'm being a noob, but I'm still not clear on how to connect the dots.  Since this is probably something most people using the package will want to do, would it be possible to have an example of creating a pdf and saving it directly to CollectionFS (preferably without saving it directly to the filesystem) added to the README.md?

**Comment by kartikTT** [comment]
Was there ever a resolution to this? Have exhausted many options with trying to serve pdf on client. An example of creating a pdf and saving it to CollectionFS would be incredibly useful.

**Comment by jaredtmartin** [comment]
Yep, I'm still fighting with this too.  It would also be nice to see how to return the file to client without using CollectionFS or Iron Router. Especially for when you want to generate pdfs from servers hosted at meteor.com

**Comment by jaredtmartin** [comment]
In the snippet above, they did:
fileObj = new FS.File(filePath)
FileStorage.insert fileObj

How can I create a FS.File object from the PDFDocument without touching the filesystem? (as is necessary on a project hosted on meteor.com) Is a PDFDocument a FS.File? Is it a stream?
I'd be more than willing to create some simple examples for you, but first I have to be able to do it :-)

**Comment by jaredtmartin** [comment]
Nope, I guess not:
var doc = new PDFDocument({size: 'A4', margin: 50});
doc.text('PDFKit is simple', 10, 30, {align: 'center', width: 200});
fileObj = new FS.File(doc)
Error: DataMan constructor received data that it doesn't support
    at new DataMan (packages/cfs:data-man/server/data-man-api.js:67:1)
    at setData (packages/cfs:file/fsFile-common.js:99:1)
    at EventEmitter.fsFileAttachData [as attachData] (packages/cfs:file/fsFile-common.js:94:1)
    at EventEmitter.FS.File (packages/cfs:file/fsFile-common.js:17:1)
    at repl:1:12
    at /Users/jaredmartin/Programs/returnvisits/.meteor/local/build/programs/server/shell.js:169:23

**Comment by jaredtmartin** [comment]
OK! I have finally figured it out! (I can't believe it took so long, cause it's really not that complicated)

Here's the code:
    var file = new FS.File();
    file.attachData(doc.outputSync(), {type: 'application/pdf'});
    file.name('myPDF.pdf');
    var datasheet = Datasheets.insert(file);
    return datasheet._id;

That's it.  The attachData function accepts a callback, but it's not useful to me. (I don't think)
Now, its a little tricky when you get back to the client, because event though you have the file in your collection, it still is being processed, so file.url() will return nothing.  So, in your client, you'll have to have it wait a bit.  Here's how I did it:
    Meteor.call('generate_datasheet', Session.get('postId'), function(error, result){
      if (error) console.log('There was a problem generating the datasheet PDF. ' + error);
      else {
        Tracker.autorun(function(){
          var file = Datasheets.findOne(result);
          if (file.isUploaded() && file.url()) window.open(file.url(),'_blank');
        });
      }
    });

I still don't really know how to return a PDF straight to the client without Iron Router or CollectionFS, but I'm happy with this victory.
I'm going to leave this ticket open because the request is that an example of this be added to the docs.  This was waaaayy too hard to find, and I think it's a very common case.  Either my solution or some better solution should be added to the docs.
